### PR TITLE
feat: `breadcrumb` component

### LINKED ISSUE #395: [Feature]: Breadcrumb component
### Describe the feature

New component on https://ui.shadcn.com/docs/components/breadcrumb

### Additional information

- [ ] I intend to submit a PR for this feature.
- [ ] I have already implemented and/or tested this feature.

### PR DESCRIPTION
**Pull request by Saeid-Za**
resolves #395

### CODE REVIEW

**Review by sadeghbarati — APPROVED** [review]
❤️

File: apps/www/src/lib/registry/default/ui/breadcrumb/BreadcrumbLink.vue [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  asChild?: boolean
+}>()
+
+const Comp = computed(() => props.asChild ? Slot : 'a')
+</script>
+
+<template>
+  <Comp :class="cn('transition-colors hover:text-foreground', props.class)">
+    <slot />
+  </Comp>
```
Comment (zernonia) [inline apps/www/src/lib/registry/default/ui/breadcrumb/BreadcrumbLink.vue]: Let's use `Primitive` component, and `PrimitiveProps` prop type ya

Comment (Saeid-Za) [reply apps/www/src/lib/registry/default/ui/breadcrumb/BreadcrumbLink.vue]: Thanks for the review, Done.

**Comment by sadeghbarati** [comment]
@Saeid-Za Hey

`PrimitiveProps` already has `as` type inside of it, so I just updated your code to prevent duplicated interface

**Review by zernonia — APPROVED** [review]
LGTM! Thanks for the PR @Saeid-Za ! 💚

**Pull request merged**